config-model/src/main/java/com/yahoo/vespa/model/container/Container.java

```diff
@@ -89,19 +87,24 @@ protected Container(AbstractConfigProducer parent, String name, boolean retired,
         this.index = index;
 
         if (getHttp() == null) {
-            // TODO Vespa 8: set to 1. The second (health) port has not been used since Vespa 6 or earlier.
-            numHttpServerPorts = 2;
             addChild(defaultHttpServer);
-        } else if (getHttp().getHttpServer() == null) {
-            numHttpServerPorts = 0;
-        } else {
-            numHttpServerPorts = getHttp().getHttpServer().getConnectorFactories().size();
         }
         addBuiltinHandlers();
 
         addChild(new SimpleComponent("com.yahoo.container.jdisc.ConfiguredApplication$ApplicationContext"));
     }
 
+    protected int getNumHttpServerPorts() {
+        if (getHttp() == null) {
+            // TODO Vespa 8: set to 1. The second (health) port has not been used since Vespa 6 or earlier.
+            return defaultHttpServer.getConnectorFactories().size()+1;
```

Comment:
adding +1 is only correct before initDefaultJettyConnector has added one, but this method needs to work both before and after that :-(